src/dev/RedundantADC.cpp

```diff
@@ -0,0 +1,82 @@
+#include <EVT/io/ADC.hpp>
+#include <dev/RedundantADC.hpp>
+
+namespace IO = EVT::core::IO;
+
+constexpr uint32_t LOW_MARGIN = 1;
+constexpr uint32_t HIGH_MARGIN = 5;
+
+namespace HIB::DEV {
+
+/*
+ * Constructs a RedundantADC object with the given ADC instances.
+*/
+RedundantADC::RedundantADC(IO::ADC& adc0, IO::ADC& adc1, IO::ADC& adc2) : adc0(adc0), adc1(adc1), adc2(adc2) {}
+
+/*
+ * Process readings from the ADCs and check for redundancy.
+ *
+ * This function reads values from three ADCs, performs bitwise comparison,
+ * and checks for redundancy among the readings.
+ */
+RedundantADC::Status RedundantADC::readVoltage(uint32_t& return_val) {
+    // Read ADC values
+    int32_t adcValues[3];
+    adcValues[0] = static_cast<int32_t>(adc0.read() * 1000);
+    adcValues[1] = static_cast<int32_t>(adc1.read() * 1000);
+    adcValues[2] = static_cast<int32_t>(adc2.read() * 1000);
+
+    // Calculate average of all ADC values
+    uint32_t average = (adcValues[0] + adcValues[1] + adcValues[2]) / 3;
+
+    // Check for margin error
+    bool adc0underLow = (adcValues[0] / average) < LOW_MARGIN;
+    bool adc1underLow = (adcValues[1] / average) < LOW_MARGIN;
+    bool adc2underLow = (adcValues[2] / average) < LOW_MARGIN;
+
+    bool adc0underHigh = (adcValues[0] / average) < HIGH_MARGIN;
+    bool adc1underHigh = (adcValues[1] / average) < HIGH_MARGIN;
+    bool adc2underHigh = (adcValues[2] / average) < HIGH_MARGIN;
+
+    // Check for redundancy
+    bool allUnderLow = adc0underLow && adc1underLow && adc2underLow;
+    bool allUnderHigh = adc0underHigh && adc1underHigh && adc2underHigh;
+
+    if (allUnderLow) {
+        return_val = average;
+        return RedundantADC::Status::OK;
+    } else if (allUnderHigh) {
+        return_val = 0;
```

Comment:
If they're all under the high margin, that's a margin error. We still want to return the actual value
```suggestion
        return_val = average;
```